bin/opm

```diff
@@ -2302,7 +2302,7 @@ sub trim_curl_out ($) {
     my $s = shift;
 
     if (defined $s) {
-        $s =~ s{\AHTTP/\d+(?:\.\d+)? 200 Connection established\r\n.*?\r\n\r\n}{}msi;
+        $s =~ s{\AHTTP/\d+(?:\.\d+)? 200 Connection established.*?\r\n\r\n}{}msi;
```

Comment:
Your proxy server does not put `\r\n` after the `200 Connection established` line? What does it use? A single `\n`?

Reply:
Maybe better expressed as `$s =~ s{\AHTTP/\d+(?:\.\d+)? 200 Connection established\r\n(?:.*\r\n)?\r\n}{}msi;` to convey optional additional line(s)?

Reply:
@kalahari That regex looks wrong to me. `.*` is greedy and will consume the header of the subsequent response headers (note: there are 2 responses in series here).

Reply:
@agentzh Apologies if I am mistaken, but I think it will still work, as the pattern must end on an empty line https://regex101.com/r/sMfdG9/1

Reply:
@kalahari The second response header also ends with an empty line. That does not change anything. I suggest you use real proxies and real responses to test things here. I want to see real output from real proxy servers.

Reply:
@agentzh Proxy server returns

```
HTTP/1.1 200 Connection established\r\n
\r\n
HTTP/1.1 302 Moved Temporarily\r\n
Server: openresty+\r\n
Date: Thu, 31 Aug 2017 07:11:37 GMT\r\n
Content-Type: text/html\r\n
...
```

The first `\r\n` in pattern `\r\n.*?\r\n\r\n` is redundant.

Reply:
@zhzhliang OK. I see what's going on here. Your first proxy response does not return any response header entires at all. Just the status line.